arrows/vxl/pixel_feature_extractor.cxx

```diff
@@ -0,0 +1,321 @@
+  vxl::average_frames average_frames_filter;
+  vxl::convert_image convert_image_filter;
+  vxl::color_commonality_filter color_commonality_filter;
+  vxl::high_pass_filter high_pass_bidir_filter;
+  vxl::high_pass_filter high_pass_box_filter;
+};
+
+// ----------------------------------------------------------------------------
+template < typename pix_t >
+vil_image_view< pix_t >
+pixel_feature_extractor::priv
+::concatenate_images( std::vector< vil_image_view< pix_t > > filtered_images )
+{
+  // Count the total number of planes
+  unsigned total_planes{ 0 };
+
+  for( auto const& image : filtered_images )
+  {
+    total_planes += image.nplanes();
+  }
+
+  if( total_planes == 0 )
+  {
+    LOG_ERROR( p->logger(), "No filtered images provided" );
+    return {};
+  }
+
+  auto const ni = filtered_images.at( 0 ).ni();
+  auto const nj = filtered_images.at( 0 ).nj();
+  vil_image_view< pix_t > concatenated_planes{ ni, nj, total_planes };
+
+  // Concatenate the filtered images into a single output
+  unsigned current_plane{ 0 };
+
+  for( auto const& image : filtered_images )
+  {
+    for( unsigned i{ 0 }; i < image.nplanes(); ++i )
+    {
+      vil_plane( concatenated_planes,
+                 current_plane ).deep_copy( vil_plane( image, i ) );
+      ++current_plane;
+    }
+  }
+  return concatenated_planes;
+}
+
+// ----------------------------------------------------------------------------
+template < typename pix_t >
+vil_image_view< pix_t >
+pixel_feature_extractor::priv
+::filter( kwiver::vital::image_container_sptr input_image )
+{
+  std::vector< vil_image_view< vxl_byte > > filtered_images;
+
+  if( enable_color || enable_gray )
+  {
+    const auto vxl_image = vxl::image_container::vital_to_vxl( input_image->get_image() );
+    // 3 channels
+    if( enable_color ){
+      filtered_images.push_back( vxl_image );
+    }
+
+    // 1 channel
+    if( enable_gray )
+    {
+      // TODO check whether this should be vil_convert_to_grey_using_rgb_weighting
+      filtered_images.push_back( vil_convert_to_grey_using_average( vxl_image ) );
+    }
+  }
+
+  if( enable_color_commonality )
+  {
+    // 1 channel
+    auto color_commonality = color_commonality_filter.filter( input_image );
+    filtered_images.push_back(
+        vxl::image_container::vital_to_vxl( color_commonality->get_image() ) );
+  }
+  // TODO this should also have one which does the other sort of filtering
+  if( enable_high_pass_box )
+  {
+    auto high_pass_box = high_pass_box_filter.filter( input_image );
+    // 2 channels
+    filtered_images.push_back(
+      vxl::image_container::vital_to_vxl( high_pass_box->get_image() ) );
+  }
+  if( enable_high_pass_bidir )
+  {
+    auto high_pass_bidir = high_pass_bidir_filter.filter( input_image );
+    // 2 channels
+    filtered_images.push_back(
+      vxl::image_container::vital_to_vxl( high_pass_bidir->get_image() ) );
+  }
+  if( enable_convert )
+  {
+    // 3 channels
+    // This should be removed
```

Comment:
If it should be removed, why is it here for now?

Reply:
For some reason the `convert_color` filter was in the original big PR even though it's not used in the default configs. I just removed it, but I could see the argument for leaving it in, in case we might use it in the future.